internal/go/skycfg/proto_message.go

```diff
@@ -357,6 +361,16 @@ func valueFromStarlark(t reflect.Type, sky starlark.Value) (reflect.Value, error
 			val.Elem().Set(reflect.ValueOf(sky.msg).Elem())
 			return val.Elem(), nil
 		}
+
+		dpb, ok := sky.msg.(*types.Duration)
+		if ok && t == reflect.TypeOf(time.Duration(0)) {
+			d, err := types.DurationFromProto(dpb)
+			if err != nil {
+				return reflect.Value{}, fmt.Errorf("TypeError: %v (type `%s') can't be coverted to type `%s'.", dpb, reflect.TypeOf(dpb), reflect.TypeOf(t))
```

Comment:
This would be a `ValueError`, not `TypeError`. The types match, but the value is out of range or invalid. I think you'll also want to include the error returned by [`ptypes.Duration()`](https://godoc.org/github.com/golang/protobuf/ptypes#Duration), it will tell the user what went wrong.

```go
d, err := ptypes.Duration(dpb)
if err != nil {
  return reflect.Value{}, fmt.Errorf("ValueError: %v", err)
}
```

Reply:
Ah yes, good catch, thanks.